File: sample/dotnet/BestsellerApi.ApiService/Entities/Stock.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BestsellerApi.ApiService.Entities;

[Table("stock")]
public class Stock
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [Column("item_id")]
    public string ItemId { get; set; } = string.Empty;

    [Required]
    [Column("in_stock")]
    public bool InStock { get; set; }

    [Required]
    [Column("quantity")]
    public int Quantity { get; set; }

    [Column("warehouse")]
    public string? Warehouse { get; set; }

    [Column("last_updated")]
    public DateTimeOffset? LastUpdated { get; set; }

    [Column("created_at")]
    public DateTimeOffset CreatedAt { get; set; }

    [Column("updated_at")]
    public DateTimeOffset UpdatedAt { get; set; }
}

File: sample/java/src/main/java/com/bestseller/api/entity/Stock.java
package com.bestseller.api.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.Instant;

@Table("stock")
public record Stock(
    @Id
    @Column("id")
    Integer id,

    @Column("item_id")
    String itemId,

    @Column("in_stock")
    Boolean inStock,

    @Column("quantity")
    Integer quantity,

    @Column("warehouse")
    String warehouse,

    @Column("last_updated")
    Instant lastUpdated,

    @Column("created_at")
    Instant createdAt,

    @Column("updated_at")
    Instant updatedAt
) {
}
